x/datapool/keeper/pool.go

```diff
@@ -211,9 +210,25 @@ func (k Keeper) CreatePool(ctx sdk.Context, curator sdk.AccAddress, poolParams t
 	// store pool
 	k.SetPool(ctx, newPool)
 
+	// mint tokens as many as targetNumData
+	k.setInitialSupply(ctx, poolID)
+	err = k.mintPoolShareToken(ctx, poolID, poolParams.TargetNumData)
+	if err != nil {
+		return 0, sdkerrors.Wrapf(err, "failed to mint share token")
+	}
```

Comment:
We should keep in mind these two functions (set supply and mint tokens) must be always called together. If someone changes the supply and forgets to mint tokens, the cosmos-sdk will raise an invariant error that the total supply of the token is not matched with the actual minted amount.
To prevent this, we may define one function that not only sets supply but also mints tokens.

Reply:
Ah. I just saw a related comment at the next PR #302: https://github.com/medibloc/panacea-core/pull/302#discussion_r850167499.
IIRC, whenever we mint more tokens, we have to increase the total supply of that token. So, I think the `setInitialSupply` and the `mintPoolShareToken` should be always called together. I'm not sure since I've read this part of codes long time ago.

We can test this scenario by:
- minting more tokens without increasing its supply
- dumping a new `genesis.json` by `panacead export ...`
- validating the `genesis.json` by `panacead validate-genesis ...`.

Reply:
Ah. Please forget about comments above. I haven't read the `setInitialSupply()` seriously that sets the supply to 0, not to the `TargetNumData`. Sorry for confusions.